### PR TITLE
converted migrations_table class component to functional component

### PR DESCRIPTION
**Pull request by ezekielchow**
#### Summary

Refactor `webapp/channels/src/components/admin_console/database/migrations_table.tsx` from class component into functional component

#### Ticket Link

Fixes [#27892](https://github.com/mattermost/mattermost/issues/27892)
Jira [#60070](https://mattermost.atlassian.net/browse/MM-60070)

#### Release Note

```release-note
NONE
```

### CODE REVIEW

File: webapp/channels/src/components/admin_console/database/migrations_table.tsx [hunk 1]
```diff
@@ -11,80 +11,74 @@ import type {ActionResult} from 'mattermost-redux/types/actions';
...
-                >
-                    <td className='whitespace--nowrap'>{migration.version}</td>
-                    <td className='whitespace--nowrap'>{migration.name}</td>
-                </tr>
-            );
-        });
+        handleGetAppliedSchemaMigrations();
+    });
 
+    const items = migrations.map((migration) => {
```
Comment (larkox) [inline webapp/channels/src/components/admin_console/database/migrations_table.tsx:38]: Just to not have to do the map on every render, let's use a `useMemo` here.

File: webapp/channels/src/components/admin_console/database/migrations_table.tsx [hunk 2]
```diff
@@ -11,80 +11,74 @@ import type {ActionResult} from 'mattermost-redux/types/actions';
...
+                    <tbody>
+                        {items}
+                    </tbody>
+                </table>
+            </div>
+        </div>
+    );
+};
 
 export default MigrationsTable;
```
Comment (larkox) [inline webapp/channels/src/components/admin_console/database/migrations_table.tsx:84]: This used to be a Pure Component, so add here `React.memo`.

**Review by larkox — CHANGES_REQUESTED** [review]
Great work! Just a couple of improvements.

**Comment by ezekielchow** [comment]
> Great work! Just a couple of improvements.

Updated

**Review by devinbinnie — APPROVED** [review]
LGTM! Thanks @ezekielchow!

**Comment by ezekielchow** [comment]
> Great work! Just a couple of improvements.

Updated. Added tests too since there wasn't any.

File: webapp/channels/src/components/admin_console/database/migrations_table.test.tsx [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+            ...baseProps,
+            actions: {
+                getAppliedSchemaMigrations: (jest.fn()).mockResolvedValue({
+                    data: mockMigrations,
+                } as ActionResult),
+            },
+        };
+
+        act(() => {
+            view = render(withIntl(<MigrationsTable {...withDataProps}/>));
```
Comment (larkox) [inline webapp/channels/src/components/admin_console/database/migrations_table.test.tsx:64]: You can use `renderWithContext` to get intl, and a few other things that might be needed.

File: webapp/channels/src/components/admin_console/database/migrations_table.test.tsx [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+    //     expect(actions.getUserAudits).toHaveBeenCalledTimes(1);
+    // });
+
+    // test('should hide', async () => {
+    //     render(withIntl(<AccessHistoryModal {...baseProps} />));
+    //     await waitFor(() => screen.getByText('Access History'));
+    //     fireEvent.click(screen.getByLabelText('Close'));
+    //     await waitForElementToBeRemoved(() => screen.getByText('Access History'));
+    // });
+});
```
Comment (larkox) [inline webapp/channels/src/components/admin_console/database/migrations_table.test.tsx:124]: You can remove these.

**Review by larkox — CHANGES_REQUESTED** [review]
Thank you very much for the tests! Just a couple of details.

File: webapp/channels/src/components/admin_console/database/migrations_table.test.tsx (hunk 4 above)
Comment (ezekielchow) [reply webapp/channels/src/components/admin_console/database/migrations_table.test.tsx]: Oh no. This is embarrassing

File: webapp/channels/src/components/admin_console/database/migrations_table.test.tsx (hunk 3 above)
Comment (ezekielchow) [reply webapp/channels/src/components/admin_console/database/migrations_table.test.tsx]: Updated

**Review by larkox — APPROVED** [review]
Looks good to me!

**Review by yasserfaraazkhan — APPROVED** [review]

**Pull request merged**